From: Rob Evers <revers@redhat.com>
Date: Wed, 27 Oct 2010 15:22:53 -0400
Subject: [scsi] lpfc: fix possible crash on non-SLI4 hba
Message-id: <1288192999-24221-2-git-send-email-revers@redhat.com>
Patchwork-id: 28928
O-Subject: [RHEL5.6 PATCH 01/27] lpfc: Fixed a potential driver bug which can
	cause system crash on non SLI4 hba. (CR: 105946)
Bugzilla: 639028

lpfc: Fixed a potential driver bug which can cause system crash on non SLI4 hba. (CR: 105946)

From: Rob Evers on behalf of Emulex <revers@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=636289

diff --git a/drivers/scsi/lpfc/lpfc_hbadisc.c b/drivers/scsi/lpfc/lpfc_hbadisc.c
index 657e89e..2e398b7 100644
--- a/drivers/scsi/lpfc/lpfc_hbadisc.c
+++ b/drivers/scsi/lpfc/lpfc_hbadisc.c
@@ -699,7 +699,7 @@ lpfc_work_done(struct lpfc_hba *phba)
 							(status &
 							 HA_RXMASK));
 		}
-		if (pring->txq_cnt)
+		if ((phba->sli_rev == LPFC_SLI_REV4) && pring->txq_cnt)
 			lpfc_drain_txq(phba);
 		/*
 		 * Turn on Ring interrupts
